src/OrchardCore.Modules/OrchardCore.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -391,19 +392,16 @@ private async Task GenerateContainerPathFromPattern(AutoroutePart part)
                     ContentItem = part.ContentItem
                 };
 
-                // We keep the current culture as a reference for later
-                var currentCulture = CultureInfo.CurrentUICulture;
-
-                _contentManager ??= _serviceProvider.GetRequiredService<IContentManager>();
-                var cultureAspect = await _contentManager.PopulateAspectAsync(part.ContentItem, new CultureAspect());
-
-                CultureInfo.CurrentUICulture = cultureAspect.Culture;
+                using (var cultureScope = new CultureScope(CultureInfo.CurrentUICulture))
+                {
+                    _contentManager ??= _serviceProvider.GetRequiredService<IContentManager>();
+                    var cultureAspect = await _contentManager.PopulateAspectAsync(part.ContentItem, new CultureAspect());
 
-                part.Path = await _liquidTemplateManager.RenderAsync(pattern, NullEncoder.Default, model,
-                    scope => scope.SetValue("ContentItem", model.ContentItem));
+                    CultureInfo.CurrentUICulture = cultureAspect.Culture;
```

Comment:
What's the point of creating a helper to do that, and now you set it manually. Why not call the aspect before and use the scope with the result of the aspect?

Reply:
You are right that's why you saw CurrentUICulture to CurrentUICulture ;)